sdk/client/engine/src/lib.rs

```diff
@@ -71,8 +71,41 @@ impl Engine {
     }
 }
 
-fn result_to_json_ptr<T: Serialize>(result: T) -> *mut c_char {
-    let json_string = serde_json::to_string(&result).unwrap();
+#[derive(Serialize)]
+struct FFIResponse<T> where T: Serialize {
+    response_code: ResponseCode,
```

Comment:
```suggestion
    status: Status,
```
what about renaming to status? `ResponseCode` makes me think of HTTP response code

also wondering if we even actually need status/response code, since either there is a `result` or there is an `error_message`

or is the idea that the caller could first check the `status/response_code` to determine if they want to access the `result` or `error_message` field?

Reply:
@markphelps Exactly that. I was thinking they could check that before accessing the other fields, but down for whatever makes sense.